commit 805a938e44dde07f6c737b0e8126bfa976cfa580
Author: Aaron Kennedy <aaron.kennedy@jollamobile.com>
Date:   Fri May 31 20:13:24 2013 +1000

    Lipstick platform

diff --git a/passthroughserver/main.cpp b/passthroughserver/main.cpp
index 5594278..25c080f 100644
--- a/passthroughserver/main.cpp
+++ b/passthroughserver/main.cpp
@@ -23,6 +23,9 @@
 #ifdef HAVE_WAYLAND
 #include "waylandplatform.h"
 #endif // HAVE_WAYLAND
+#ifdef HAVE_LIPSTICK
+#include "lipstickplatform.h"
+#endif // HAVE_LIPSTICK
 #include "unknownplatform.h"
 
 #include <QGuiApplication>
@@ -96,6 +99,9 @@ QSharedPointer<MInputContextConnection> createConnection(const MImServerConnecti
 
 QSharedPointer<Maliit::AbstractPlatform> createPlatform()
 {
+#ifdef HAVE_LIPSTICK
+    return QSharedPointer<Maliit::AbstractPlatform>(new Maliit::LipstickPlatform);
+#endif
 #ifdef HAVE_WAYLAND
     if (QGuiApplication::platformName() == "wayland") {
         return QSharedPointer<Maliit::AbstractPlatform>(new Maliit::WaylandPlatform);
diff --git a/passthroughserver/passthroughserver.pro b/passthroughserver/passthroughserver.pro
index 373a4ce..d7af09e 100644
--- a/passthroughserver/passthroughserver.pro
+++ b/passthroughserver/passthroughserver.pro
@@ -23,6 +23,8 @@ for(OPTION,$$list($$lower($$COV_OPTION))){
     }
 }
 
+lipstick:DEFINES+=HAVE_LIPSTICK
+
 QMAKE_CLEAN += *.gcno *.gcda
 
 INSTALLS += target
diff --git a/src/lipstickplatform.cpp b/src/lipstickplatform.cpp
new file mode 100644
index 0000000..f380f42
--- /dev/null
+++ b/src/lipstickplatform.cpp
@@ -0,0 +1,36 @@
+/* * This file is part of Maliit framework *
+ *
+ * Copyright (C) 2013 Jolla Ltd.
+ *
+ * Contact: maliit-discuss@lists.maliit.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.LGPL included in the packaging
+ * of this file.
+ */
+
+#include "lipstickplatform.h"
+
+#include <QGuiApplication>
+#include <qpa/qplatformnativeinterface.h>
+
+namespace Maliit
+{
+
+void LipstickPlatform::setupInputPanel(QWindow* window, Maliit::Position)
+{
+    QPlatformNativeInterface *native = QGuiApplication::platformNativeInterface();
+    window->create();
+    native->setWindowProperty(window->handle(), QLatin1String("CATEGORY"), QString("overlay"));
+}
+
+void LipstickPlatform::setInputRegion(QWindow* window, const QRegion& region)
+{
+    QPlatformNativeInterface *native = QGuiApplication::platformNativeInterface();
+    window->create();
+    native->setWindowProperty(window->handle(), QLatin1String("MOUSE_REGION"), QVariant(region));
+}
+
+} // namespace Maliit
diff --git a/src/lipstickplatform.h b/src/lipstickplatform.h
new file mode 100644
index 0000000..226551a
--- /dev/null
+++ b/src/lipstickplatform.h
@@ -0,0 +1,30 @@
+/* * This file is part of Maliit framework *
+ *
+ * Copyright (C) 2013 Jolla Ltd.
+ *
+ * Contact: maliit-discuss@lists.maliit.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.LGPL included in the packaging
+ * of this file.
+ */
+
+#ifndef MALIIT_LIPSTICK_PLATFORM_H
+#define MALIIT_LIPSTICK_PLATFORM_H
+
+#include "abstractplatform.h"
+
+namespace Maliit
+{
+
+class LipstickPlatform : public AbstractPlatform
+{
+    virtual void setupInputPanel(QWindow* window, Maliit::Position position);
+    virtual void setInputRegion(QWindow* window, const QRegion& region);
+};
+
+} // namespace Maliit
+
+#endif // MALIIT_UNKNOWN_PLATFORM_H
diff --git a/src/src.pro b/src/src.pro
index b1f7c76..dc0b6e7 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -86,6 +86,11 @@ wayland {
     SERVER_SOURCES += waylandplatform.cpp
 }
 
+lipstick {
+    SERVER_HEADERS_PRIVATE += lipstickplatform.h
+    SERVER_SOURCES += lipstickplatform.cpp
+}
+
 SETTINGS_HEADERS_PRIVATE += \
         mimsettingsqsettings.h \
         mimsettings.h \
